### ISSUE TITLE
ImageGrid in 3.0

### ISSUE THREAD

**Issue by Phlya** [matplotlib/matplotlib#12505]
### Bug report

**Bug summary**

ImageGrid seems to be broken on 3.0.0?

**Code for reproduction**

```python
f = plt.figure()
grid = ImageGrid(f, 111, 
                 nrows_ncols=(2, 2),
                 direction='column',
                 label_mode="L",
                 cbar_location="right",
                 cbar_mode="single",
                 cbar_size="5%",
                 cbar_pad="3%"
                 )
axarr = np.array(grid).reshape(2, 2)
for ax in axarr.flatten():
    m = ax.imshow(np.random.random(15**2).reshape((15, 15))+0.5, 
           norm=LogNorm(vmax=2, vmin=0.5), cmap='coolwarm')
plt.colorbar(m, cax=grid.cbar_axes[0], format=FormatStrFormatter('%.2f'))
```

**Actual outcome**
![image](https://user-images.githubusercontent.com/2895034/46876325-f1cafa00-ce35-11e8-955e-c4ae429dd9c0.png)

**Expected outcome**
From 2.2.2
![image](https://user-images.githubusercontent.com/2895034/46876255-c3e5b580-ce35-11e8-837e-5880a42bf4f4.png)

Note: if I remove all colorbar-related code, it again looks weird, but in a different way:

```python
f = plt.figure()
grid = ImageGrid(f, 111,  
                 nrows_ncols=(2, 2),
                 direction='column',
                 label_mode="L",
                 )
axarr = np.array(grid).reshape(2, 2)
for ax in axarr.flatten():
    m = ax.imshow(np.random.random(15**2).reshape((15, 15))+0.5, 
           norm=LogNorm(vmax=2, vmin=0.5), cmap='coolwarm')
```
![image](https://user-images.githubusercontent.com/2895034/46876557-89304d00-ce36-11e8-88ab-3f22f63941b6.png)

Seems like maybe all axes take up the whole figure, and so cover the rest?.. Just a guess.

**Matplotlib version**

  * Operating system: Ubuntu 16.04
  * Matplotlib version: 3.0.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): inline
  * Python version: 3.5.2
  * Jupyter version (if applicable): Jupyter Lab 0.33.6

**Comment by jklymak** [comment]
Duplicate #12355.  See fix in #12363

**Comment by tacaswell** [comment]
Great!  Just merged #12363
